cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -152,8 +269,24 @@ impl ClientTrait for MockClient {
     async fn resolve(
         &self,
         _package_groups: Vec<PackageGroup>,
-    ) -> Result<Vec<ResolvedPackageGroup>, ResolveError> {
-        unimplemented!()
+    ) -> Result<ResolvedGroups, ResolveError> {
+        let mock_resp = self
+            .mock_responses
+            .lock()
+            .expect("couldn't acquire mock lock")
+            .pop_front();
```

Comment:
nit (nb): for future debugging purposes we might want to use a vec + index (or cursor?) instead to provide better diagnostics which response was provided.

Reply:
That sounds easy enough to implement if we decide at some point in the future that it's useful.